provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java

```diff
@@ -99,6 +103,69 @@ public SecurityProviderTPMEmulator(String registrationId) throws SecurityProvide
         srkPublic = createPersistentPrimary(tpm, SRK_PERSISTENT_HANDLE, TPM_RH.OWNER, SRK_TEMPLATE, "SRK");
     }
 
+    /**
+     * Constructor for creating a Security Provider on TPM Simulator with the supplied Registration ID
+     * @param registrationId A non {@code null} or empty value tied to this registration
+     * @throws SecurityProviderException If the constructor could not start the TPM
+     */
+    public SecurityProviderTPMEmulator(String registrationId, int tpmConnectRetryAttempts) throws SecurityProviderException
+    {
+        if (registrationId == null || registrationId.isEmpty())
+        {
+            throw new IllegalArgumentException("Registration Id cannot be null or empty");
+        }
+        if (!registrationId.matches(REGEX_FOR_VALID_REGISTRATION_ID))
+        {
+            throw new IllegalArgumentException("The registration ID is alphanumeric, lowercase, and may contain hyphens. Max characters allowed is 128.");
+        }
+
+        if (tpmConnectRetryAttempts <= 0)
```

Comment:
same here: add reference here as a TODO item so that we can remove the retry after e2e issue on tpm is fixed.